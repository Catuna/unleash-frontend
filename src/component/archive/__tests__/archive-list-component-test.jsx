import React from 'react';

import ArchiveList from '../archive-list-component';
import renderer from 'react-test-renderer';

jest.mock('react-mdl');
// TODO mock DropdownButton
// jest.mock('../../common', () => ({ DropdownButton: 'DropdownButton', styles: {} }));
const archive = [
    {
        name: 'adin-pay-confirm-disabled',
        description: 'Disables the confirm-functionality from API',
        enabled: false,
        strategies: [{ name: 'default', parameters: {} }],
        createdAt: '2016-10-25T15:38:28.573Z',
        reviveName: 'adin-pay-confirm-disabled',
    },
    {
        name: 'adin-pay-platform-sch-payment',
        description: 'Enables use of schibsted payment from order-payment-management',
        enabled: true,
        strategies: [{ name: 'default', parameters: {} }],
        createdAt: '2016-08-03T12:41:35.631Z',
        reviveName: 'adin-pay-platform-sch-payment',
    },
];

xtest('renders correctly with no archived toggles', () => {
    const featureMetrics = { lastHour: {}, lastMinute: {}, seenApps: {} };
    const settings = {
        feature: {
            filter: '',
            sort: 'name',
            showLastHour: false,
        },
    };

    const tree = renderer
        .create(
            <ArchiveList
                name={'ff'}
                fetchArchive={jest.fn()}
                archive={[]}
                settings={settings}
                featureMetrics={featureMetrics}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

xtest('renders correctly with archived toggles', () => {
    const featureMetrics = { lastHour: {}, lastMinute: {}, seenApps: {} };
    const settings = {
        feature: {
            filter: '',
            sort: 'name',
            showLastHour: false,
        },
    };

    const tree = renderer
        .create(
            <ArchiveList
                name={'ff'}
                fetchArchive={jest.fn()}
                archive={archive}
                settings={settings}
                featureMetrics={featureMetrics}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
