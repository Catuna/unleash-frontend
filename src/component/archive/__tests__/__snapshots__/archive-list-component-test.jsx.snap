// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with archived toggles 1`] = `
<react-mdl-Card
  className="fullwidth"
  shadow={0}
>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <react-mdl-List>
        <react-mdl-ListItem
          className="archiveList"
        >
          <span
            className="listItemToggle"
          >
            Toggle name
          </span>
          <span
            className="listItemRevive"
          >
            Revive
          </span>
        </react-mdl-ListItem>
        <hr />
        <react-mdl-List>
          <react-mdl-ListItem
            twoLine={true}
          >
            <react-mdl-ListItemAction>
              <react-mdl-Icon
                name="keyboard_arrow_right"
              />
            </react-mdl-ListItemAction>
            <react-mdl-ListItemContent>
              <a
                className="listLink truncate"
                onClick={[Function]}
                style={Object {}}
              >
                adin-pay-confirm-disabled
                <div
                  className="toggleDetails"
                >
                  Disables the confirm-functionality from API
                </div>
              </a>
            </react-mdl-ListItemContent>
            <react-mdl-ListItemAction
              onClick={[Function]}
            >
              <react-mdl-Icon
                name="undo"
              />
            </react-mdl-ListItemAction>
          </react-mdl-ListItem>
          <react-mdl-ListItem
            twoLine={true}
          >
            <react-mdl-ListItemAction>
              <react-mdl-Icon
                name="keyboard_arrow_right"
              />
            </react-mdl-ListItemAction>
            <react-mdl-ListItemContent>
              <a
                className="listLink truncate"
                onClick={[Function]}
                style={Object {}}
              >
                adin-pay-platform-sch-payment
                <div
                  className="toggleDetails"
                >
                  Enables use of schibsted payment from order-payment-management
                </div>
              </a>
            </react-mdl-ListItemContent>
            <react-mdl-ListItemAction
              onClick={[Function]}
            >
              <react-mdl-Icon
                name="undo"
              />
            </react-mdl-ListItemAction>
          </react-mdl-ListItem>
        </react-mdl-List>
      </react-mdl-List>
    </div>
  </div>
</react-mdl-Card>
`;

exports[`renders correctly with no archived toggles 1`] = `
<react-mdl-Card
  className="fullwidth"
  shadow={0}
>
  <div
    className="emptyState"
  >
    <react-mdl-Icon
      className="mdl-color-text--grey-300"
      name="archive"
      style={
        Object {
          "fontSize": "56px",
        }
      }
    />
    <br />
    No archived feature toggles, go see 
    <a
      onClick={[Function]}
      style={Object {}}
    >
      active toggles here
    </a>
  </div>
</react-mdl-Card>
`;
